### ISSUE TITLE
dotnet-install.sh doesn't properly detect musl systems without os-release

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/install-scripts#217]
@TinfoilSubmarine commented on [Wed Sep 15 2021](https://github.com/dotnet/source-build/issues/2459)

Not sure if this is where this issue goes, but I need to start somewhere.

The script that gets pulled in from https://dot.net/v1/dotnet-install.sh doesn't detect that my system is musl since I don't have anything at `/etc/os-release` since it just returns at 178 if it can't find os-release. Copied below:
```
    elif [ "$uname" = "Linux" ]; then
        local linux_platform_name
178 ->  linux_platform_name="$(get_linux_platform_name)" || { echo "linux" && return 0 ; }

        if [ "$linux_platform_name" = "rhel.6" ]; then
            echo $linux_platform_name
            return 0
        elif is_musl_based_distro; then
            echo "linux-musl"
            return 0
        elif [ "$linux_platform_name" = "linux-musl" ]; then
            echo "linux-musl"
            return 0
        else
            echo "linux"
            return 0
        fi
    fi
```

---

@dotnet-issue-labeler[bot] commented on [Wed Sep 15 2021](https://github.com/dotnet/source-build/issues/2459#issuecomment-920502561)

I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by AR-May** [comment]
@TinfoilSubmarine
Thank you for reaching us with this issue. You should be able to specify the os manually and workaround this problem adding ` --os "linux-musl"` to your dotnet-install call. 

I will put a question about our os detection approach for a discussion, but it seems to me as not very usual thing (that is, not to have `/etc/os-release` file).

**Comment by TinfoilSubmarine** [comment]
I ran into this issue when trying to build dotnet using source-build in a chroot that didn't have os-release created. Building in a chroot seems like a fairly common thing for packaging.

**Comment by AR-May** [comment]
Thank you for providing more details. They would be helpful for our triage!

**Comment by bekir-ozturk** [comment]
Fixed in #220
